packages/framework-core/test/booster.test.ts

```diff
@@ -81,35 +72,145 @@ describe('the `Booster` class', () => {
         })
       })
       await Booster.readModel(TestReadModel).search()
-
       expect(searcherFunctionFake).to.have.been.calledOnceWithExactly(
         match.any,
         match.any,
         TestReadModel.name,
         match.any
       )
     })
+    it('has an instance method', async () => {
+      const searcherFunctionFake = fake.returns([{ id: '42' }])
+      Booster.configureCurrentEnv((config) => {
+        replaceGetter(config, 'provider', () => {
+          return {
+            readModels: {
+              search: searcherFunctionFake,
+            },
+          } as any
+        })
+      })
+      const readModels = await Booster.readModel(TestReadModel).search()
+      for (const readModel of readModels) {
+        expect(readModel.getId()).to.not.throw
+      }
+      expect(searcherFunctionFake).to.have.been.calledOnce
+    })
+  })
+  describe('the `event` method', () => {
+    class TestEvent {
+      public constructor(readonly id: UUID) {}
+      public entityID(): UUID {
+        return this.id
+      }
+      public getId(): UUID {
+        return this.id
+      }
+    }
+    it('has an instance method', async () => {
+      const searchResult: EventSearchResponse[] = [
+        {
+          requestID: random.uuid(),
+          type: TestEvent.name,
+          entity: random.alpha(),
+          entityID: random.uuid(),
+          createdAt: random.alphaNumeric(),
+          value: {
+            id: '1',
+            entityID: () => UUID.generate(),
+          } as EventInterface,
+        },
+      ]
+      const providerEventsSearch = fake.returns(searchResult)
+      Booster.configureCurrentEnv((config) => {
+        config.provider = {
+          events: {
+            search: providerEventsSearch,
+          },
+        } as unknown as ProviderLibrary
```

Comment:
is this double `as` on purpose? 😮

Reply:
I think it is the only way to bypass the compiler in this case

Reply:
Yes, that's it. The other option would be as `any`. Thanks @alvaroloes!

Reply:
Awesome! 😄